### ISSUE TITLE
timeseries_generator  "Error when checking target: expected dense_1 to have 3 dimensions, but got array with shape (1, 10000)"

### ISSUE THREAD

**Issue by westland** [rstudio/keras#384]
I am having trouble figuring out the shape of the object that timeseries_generator in Keras/R creates for fit_generator().  I have given my generator "data" and "targets" files:
> str(data_reshp)
 num [1:681, 1:110000] 0 0 0 0 0 0 0 0 0 0 ...
> str(trgt)
 num [1:681, 1:10000] 0 0 0 0 0 0 0 0 0 0 ...

and set parameters:
lngth = 24     
samp_rate = 1  
strd = 1       
btch_sz = 1  
val_steps <- (nrow(data) - 401 - lngth) / btch_sz

train_gen = timeseries_generator(
    data=data_reshp, 
    targets=trgt, 
    length=lngth, 
    sampling_rate = samp_rate, 
    stride = strd,   
    start_index = 1, 
    end_index = 400, 
    shuffle = FALSE, 
    reverse = FALSE,
    batch_size = btch_sz
)

val_gen = timeseries_generator(
    data=data_reshp, 
    targets=trgt,
    length=lngth, 
    sampling_rate = samp_rate, 
    stride= strd,   
    start_index = 401, 
    end_index = NULL, 
    shuffle = FALSE, 
    reverse = FALSE,
    batch_size = btch_sz
)

model <- keras_model_sequential() %>%
layer_dense(units = 32, input_shape = c(lngth/samp_rate, 110000))  

model %>% compile(
    optimizer = optimizer_rmsprop(),
    loss = "mae",
    metrics = c("mae","mean_squared_logarithmic_error")
)

model %>% fit_generator(
    train_gen,
    steps_per_epoch = 10,
    epochs = 10,
    validation_data = val_gen,
    validation_steps = val_steps  
)

I get "Error when checking target: expected dense_1 to have 3 dimensions, but got array with shape (1, 10000)"

I'm presuming this is because I don't understand the definitions of   "targets=trgt, length=lngth, " parameters ... I'm not sure what shape data timeseries_generator() is generating

The definition of "targets" says that this should have the same length as data ... does this mean the same number of rows nrow() or actual length (in which case it is data: 110000 vs. targets: 10000)

I know I'm not getting something, but can't figure out what I have missed.  Any help would be greatly appreciated.
